src/components/Bag/classes.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { v4 as uuidv4 } from "uuid";
+import { Label, InsertDriveFile, Folder } from "@material-ui/icons";
+
+class BagItem {
+    constructor(item) {
+        this.item = item;
+    }
+
+    icon(t) {
+        return <Label />;
+    }
+
+    get label() {
+        return (
+            this.item?.label ||
+            this.item?.name ||
+            this.item?.description ||
+            "override label() please"
+        );
+    }
+
+    get shareable() {
+        return false;
+    }
+
+    get downloadable() {
+        return false;
+    }
+
+    get id() {
+        return this.item.id;
+    }
+}
+
+class FileBagItem extends BagItem {
+    icon(t) {
+        return <InsertDriveFile />;
+    }
+
+    get shareable() {
+        return true;
+    }
+
+    get downloadable() {
+        return true;
+    }
+
+    get name() {
+        return this.item.path.substring(this.item.path.lastIndexOf("/") + 1);
+    }
+
+    get path() {
+        return this.item.path;
+    }
+}
+
+class FolderBagItem extends BagItem {
+    icon(t) {
+        return <Folder />;
+    }
+
+    get shareable() {
+        return true;
+    }
+
+    get name() {
+        return this.item.path.substring(this.item.path.lastIndexOf("/") + 1);
+    }
+    get path() {
+        return this.item.path;
+    }
+}
+
+class AnalysisBagItem extends BagItem {
+    icon(t) {
+        return <img src="/analyses_selected.png" alt={t("common:analyses")} />;
+    }
+
+    get shareable() {
+        return true;
+    }
+}
+
+class AppBagItem extends BagItem {
+    icon(t) {
+        return <img src="/apps_selected.png" alt={t("common:apps")} />;
+    }
+
+    get shareable() {
+        return true;
+    }
+}
+
+export const FOLDER_TYPE = "folder";
+export const FILE_TYPE = "file";
+export const ANALYSIS_TYPE = "analysis";
+export const APP_TYPE = "app";
+
+export const createNewBagItem = (item) => {
+    if (!item.id) {
+        item.id = uuidv4();
```

Comment:
If the item type is anything other than data, and (for some reason, I'm not sure how this would actually happen) it doesn't have an id, I don't think creating an ID will work 🤔 I'm just thinking of sharing, because in order to share apps, analyses, or tools, the sharing endpoints require the id.